source/common/upstream/health_discovery_service.h

```diff
@@ -41,28 +120,67 @@ class HdsDelegate
       std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier>&& message) override;
   void onReceiveTrailingMetadata(Http::HeaderMapPtr&& metadata) override;
   void onRemoteClose(Grpc::Status::GrpcStatus status, const std::string& message) override;
-
+  envoy::service::discovery::v2::HealthCheckRequestOrEndpointHealthResponse sendResponse();
   // TODO(htuch): Make this configurable or some static.
-  const uint32_t RETRY_DELAY_MS = 5000;
+  // Processes the management server requests
+  std::vector<HdsClusterPtr> hdsClusters() { return hds_clusters_; };
 
 private:
-  void setRetryTimer();
-  void establishNewStream();
-  void sendHealthCheckRequest();
+  friend class HdsDelegateFriend;
+
+  void setHdsRetryTimer();
+  void setHdsStreamResponseTimer();
   void handleFailure();
+  // Establishes a connection with the management server
+  void establishNewStream();
+  void
+  processMessage(std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier>&& message);
 
   HdsDelegateStats stats_;
+  const Protobuf::MethodDescriptor& service_method_;
+
   Grpc::AsyncClientPtr async_client_;
   Grpc::AsyncStream* stream_{};
-  const Protobuf::MethodDescriptor& service_method_;
-  Event::TimerPtr retry_timer_;
-  Event::TimerPtr response_timer_;
+  Event::Dispatcher& dispatcher_;
+  Runtime::Loader& runtime_;
+  Envoy::Stats::Store& store_stats;
+  Ssl::ContextManager& ssl_context_manager_;
+  Secret::SecretManager& secret_manager_;
+  Runtime::RandomGenerator& random_;
+  ClusterInfoFactory& info_factory_;
+  AccessLog::AccessLogManager& access_log_manager_;
+
   envoy::service::discovery::v2::HealthCheckRequest health_check_request_;
   std::unique_ptr<envoy::service::discovery::v2::HealthCheckSpecifier> health_check_message_;
+
   std::vector<std::string> clusters_;
+  std::vector<HdsClusterPtr> hds_clusters_;
+
+  Event::TimerPtr hds_stream_response_timer_;
+  Event::TimerPtr hds_retry_timer_;
+
+  // How often we retry to establish a stream to the management server
+  const uint32_t RetryDelayMilliseconds = 5000;
+
+  // Soft limit on size of the cluster’s connections read and write buffers.
+  static constexpr uint32_t ClusterConnectionBufferLimitBytes = 32768;
+
+  // The timeout for new network connections to hosts in the cluster.
+  static constexpr uint32_t ClusterTimeoutSeconds = 1;
+
+  // How often envoy reports the healthcheck results to the server
+  uint32_t server_response_ms_ = 0;
 };
 
 typedef std::unique_ptr<HdsDelegate> HdsDelegatePtr;
 
+// Friend class of HdsDelegate, making it easier to access private fields
+class HdsDelegateFriend {
```

Comment:
I think you can just forward declare `class HdsDelegateFriend` and only fill in its contents in the test module.